packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -178,84 +178,98 @@ export default function QueryInspectorControls( {
 					</PanelBody>
 				</InspectorControls>
 			) }
-			{ ! inherit && (
-				<InspectorControls>
-					<ToolsPanel
-						className="block-library-query-toolspanel__filters"
-						label={ __( 'Filters' ) }
-						resetAll={ () => {
-							setQuery( {
-								author: '',
-								parents: [],
-								search: '',
-								taxQuery: null,
-							} );
-							setQuerySearch( '' );
-						} }
-					>
-						{ !! taxonomies?.length &&
-							isControlAllowed( allowedControls, 'taxQuery' ) && (
+			{ ! inherit &&
+				( isControlAllowed( allowedControls, 'taxQuery' ) ||
```

Comment:
It's better to use the same approach as `showSettingsPanel` with dedicated variables declarations for two reasons. Avoid extra calls of `isControlAllowed` and also some controls have extra checks, and in this PR we can still end up with just the panel title. For example `parents` also require the post type to be hierarchical `isPostTypeHierarchical` and for `taxQuery` we also check `!! taxonomies?.length`

Reply:
@ntsekouras thanks for having a look. I'll get this updated asap.

Reply:
Hey @ryanwelcher! Any chance you have bandwidth to update? If not I could take over in this or a new PR. Thanks!

Reply:
Thanks for the reminder @ntsekouras! I'l have this updated very shortly.